### PR TITLE
Implement a "SQL Row" type descriptor

### PR DESCRIPTION
**Pull request by 1st1**
Couple of notes:

* A dedicated descriptor is needed to reduce complexity in client libraries, as the "record" type has a different API from the "free object" type. Adding multiple cases to the codecs pipeline to track what language was used -- EdgeQL or SQL -- is more error prone and time consuming than adding a dedicated descriptor.

* This commit uses a hack to call the compiler process to build the actual descriptor. The IO<->compiler RPC code is incredibly complex and brittle, so adding two new remote methods would require copy/pasting (as well as careful testing) of a lot of code that manages compiler state and related optimizations.

  It's easier to leave all of that code alone and multiplex the new method into the existing `compile()` and `compile_in_tx()`.

  An alternative would be quite an invasive refactoring of the entire compiler_pool package, something that I'd like to avoid, especially that close to the release.

### CODE REVIEW

File: edb/server/compiler/rpc.pyx [hunk 1]
```diff
@@ -17,9 +17,12 @@
 #
 
 from typing import (
+    List,
```
Comment (elprans) [inline edb/server/compiler/rpc.pyx:20]: Just use plain `tuple` and `list`

File: edb/server/compiler/compiler.py [hunk 2]
```diff
@@ -707,6 +722,19 @@ def compile_in_tx(
     ) -> Tuple[
         dbstate.QueryUnitGroup, Optional[dbstate.CompilerConnectionState]
     ]:
+        if request.input_language is enums.InputLanguage.SQL_PARAMS:
+            tx = state.current_tx()
+            return (
```
Comment (elprans) [inline edb/server/compiler/compiler.py:727]: This doesn't type check? Can we return a proper QueryUnitGroup instead?

Comment (1st1) [reply edb/server/compiler/compiler.py:745]: I fixed all type checking issues.

**Comment by 1st1** [comment]
This PR depends on https://github.com/edgedb/edgedb-python/pull/560 and then a new version of edgedb-python

**Review by elprans — APPROVED** [review]

**Pull request merged**